api/v1beta1/metal3machine_types.go

```diff
@@ -99,9 +100,11 @@ func (s *Metal3MachineSpec) IsValid() error {
 	if err != nil {
 		invalid = append(invalid, "Image.URL")
 	}
-	_, err = url.ParseRequestURI(s.Image.Checksum)
-	if err != nil {
-		invalid = append(invalid, "Image.Checksum")
+	if s.Image.DiskFormat == nil || *s.Image.DiskFormat != "live-iso" {
+		_, err = url.ParseRequestURI(s.Image.Checksum)
```

Comment:
This change requires the checksum to be a URL, which is a breaking change and not in line with the description of the field as it also allows to provide a checksum directly.

Reply:
Also, why is this not checked on the webhook as well?

Reply:
> This change requires the checksum to be a URL

This is the original code,  I haven't changed the way `s.Image.Checksum` is validated.

Reply:
You are right, the change was introduced with this PR: https://github.com/metal3-io/cluster-api-provider-metal3/pull/604
It is still breaking though, but was introduced with Release 1.1.2.